spec/controllers/concerns/billable_event_trackable_spec.rb

```diff
@@ -21,6 +21,15 @@
   let(:ial_context) { IalContext.new(ial: 1, service_provider: current_sp) }
   let(:request_id) { SecureRandom.hex }
   let(:session_started_at) { 5.minutes.ago }
+  let(:active_profile) do
+    create(
+      :profile,
+      :active,
+      :verified,
+      user: current_user,
+      initiating_service_provider: current_sp,
```

Comment:
let's link this to a different service provider (this helps us make sure if we ever added a bug that mixed up SPs, that we'd still catch it
```suggestion
  let(:profile_sp) { create(:service_provider) }
  let(:active_profile) do
    create(
      :profile,
      :active,
      :verified,
      user: current_user,
      initiating_service_provider: profile_sp,
```